bothan-core/src/registry/processor/weighted_median.rs

```diff
@@ -0,0 +1,217 @@
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::ops::{Add, Div};
+
+use bincode::de::{BorrowDecoder, Decoder};
+use bincode::enc::Encoder;
+use bincode::error::{DecodeError, EncodeError};
+use bincode::{BorrowDecode, Decode, Encode};
+use num_traits::{FromPrimitive, Zero};
+use rust_decimal::Decimal;
+use serde::{Deserialize, Serialize};
+
+use crate::registry::processor::{Process, ProcessError};
+
+/// The `WeightedMedianProcessor` finds the weighted median of a given data set where the dataset
+/// contains the source and the value. It also has a `minimum_cumulative_weight` which is the
+/// minimum cumulative weight required to calculate the weighted median. If the cumulative weight
+/// of the data sources is less than `minimum_cumulative_weight` or the source associated with the
+/// data does not have an assigned weight, it returns an error.
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct WeightedMedianProcessor {
+    pub source_weights: HashMap<String, Decimal>,
```

Comment:
I think weight can just integer for simplicity. also make the code less error prone as we will use u32 for weight and Decimal for price (instead of now both being decimal).